.github/workflows/release.yaml

```diff
@@ -0,0 +1,219 @@
+name: PR Build
+on:
+  push:
+    tags:
+      - 'v[0-9].[0-9]+.[0-9]+'
+env:
+  GO_VERSION: 1.16.5
+jobs:
+  cache-deps:
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Setup go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ env.GO_VERSION }}
+      - name: Setup dep cache
+        uses: actions/cache@v2
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+      - name: Pull go deps
+        run: go mod download
+
+  lint:
+    runs-on: ubuntu-18.04
+    needs: cache-deps
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Setup go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ env.GO_VERSION }}
+      - name: Load cached deps
+        uses: actions/cache@v2
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+      - name: Setup build tool cache
+        uses: actions/cache@v2
+        with:
+          path: .build
+          key: ${{ runner.os }}-tools-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-tools-
+      - name: Lint
+        run: make lint
+      - name: Tidy check
+        run: make tidy-check
+      - name: Generate check
+        run: make generate-check
+      - name: Shell check
+        run: shellcheck .github/workflows/scripts/*.sh
+
+  unit-test:
+    strategy:
+      matrix:
+        OS: [ubuntu-18.04, macos-latest]
+    runs-on: ${{ matrix.OS }}
+    needs: cache-deps
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Setup go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ env.GO_VERSION }}
+      - name: Load cached deps
+        uses: actions/cache@v2
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+      - name: Run unit tests
+        run: ./.github/workflows/scripts/run_unit_tests.sh
+
+  artifacts:
+    runs-on: ubuntu-18.04
+    needs: [cache-deps]
```

Comment:
Since we know that this is now a release build ... And it is also now a different workflow in the GH UI with the "release" name and artifacts attached to it ... Should we make artifact and image build depend on unit and integration tests?

Reply:
We can't do integration tests w/o image build but regardless, I think I'd probably prefer to keep the flow the same between these? It will make it easier to adjust them all in tandem when needed (at least until GH lets us share actions more easily).